tcmalloc/internal/percpu_tcmalloc.h

```diff
@@ -357,13 +357,143 @@ static inline ABSL_ATTRIBUTE_ALWAYS_INLINE int TcmallocSlab_Push(
   int cpu = VirtualRseqCpuId();
   return f(cpu, cl, item);
 }
-#endif  // defined(__x86_64__)
+#endif // defined(__x86_64__)
+
+#if defined(__aarch64__)
+
+template <size_t Shift, size_t NumClasses>
+static inline ABSL_ATTRIBUTE_ALWAYS_INLINE int TcmallocSlab_Push(
+    typename TcmallocSlab<Shift, NumClasses>::Slabs* slabs, size_t cl,
+    void* item, OverflowHandler f) {
+  void* region_start;
+  uint64_t cpu_id;
+  void* slab_header;
+  uintptr_t current;
+  uintptr_t end;
+#if TCMALLOC_PERCPU_USE_RSEQ_ASM_GOTO
+  asm goto(
+#else
+  bool overflow;
+  asm volatile(
+#endif
+      // TODO(b/141629158):  __rseq_cs only needs to be writeable to allow for
+      // relocations, but could be read-only for non-PIE builds.
+      ".pushsection __rseq_cs, \"aw?\"\n"
+      ".balign 32\n"
+      ".local __rseq_cs_TcmallocSlab_Push_%=\n"
+      ".type __rseq_cs_TcmallocSlab_Push_%=,@object\n"
+      ".size __rseq_cs_TcmallocSlab_Push_%=,32\n"
+      "__rseq_cs_TcmallocSlab_Push_%=:\n"
+      ".long 0x0\n"
+      ".long 0x0\n"
+      ".quad 4f\n"
+      ".quad 5f - 4f\n"
+      ".quad 2f\n"
+      ".popsection\n"
+#if !defined(__clang_major__) || __clang_major__ >= 9
+      ".reloc 0, R_AARCH64_NONE, 1f\n"
```

Comment:
What is the purpose of this?

Reply:
https://reviews.llvm.org/D61973 LLVM integrated assembler since 9 started to support `R_AARCH64_NONE`. This is for establishing an artificial dependency so that linkers will not collect the referenced sections.